doc/tutorials/vuart_configuration.rst

```diff
@@ -214,3 +214,131 @@ started, as shown in the diagram below:
+        <target_vm_id>0</target_vm_id>
+        <target_uart_id>1</target_uart_id>
+      </communication_vuart>
+   </vm>
+
+The ACRN vUART related xml fields:
+ - 'id' in <legacy_vuart>, value of vuart_idx, id=0 is for legacy vuart[0] configuration, id=1 is for vuart[1].
+ - 'type' in <legacy_vuart>, type is always VUART_LEGACY_PIO for legacy vUART.
+ - 'base' in <legacy_vuart>, is use the legacy vUART port, set COM1_BASE for vuart[0], set COM2_BASE for vuart[1].INVALID_COM_BASE means do not use the legacy vUART port.
+ - 'irq' in <legacy_vuart>, is use the legacy vUART port, set COM1_IRQ for vuart[0], set COM2_IRQ for vuart[1];
+ - 'id' in <console_vuart> and <communication_vuart>, vuart_idx for PCI-vUART
+ - 'base' in <console_vuart> and <communication_vuart>, PCI_VUART means use this PCI-vUART, it is a console. INVALID_PCI_BASE means do not use this PCI-VUART.
+ - 'target_vm_id' and 'target_uart_id', connection settings for this vUART port.
+
+Run ``make BOARD_FILE=$PWD/misc/acrn-config/xmls/board-xmls/<board>.xml SCENARIO_FILE=/home/taoyuhong/acrn/acrn-hypervisor/misc/acrn-config/xmls/config-xmls/<board>/<scenario.xml``
+to build ACRN with xml configuration file. Configuration tools will test your settings, check `vUART rules <https://projectacrn.github.io/latest/tutorials/vuart_configuration.html#index-of-vuart>`_ for compile issue. After compile, you can find ``./misc/vm_configs/scenarios/<scenario>/<board>/pci_dev.c`` has been changed by configuration tools, something like:
+
+.. code-block:: none
+
+   {
+      .emu_type = PCI_DEV_TYPE_HVEMUL,
+      .vbdf.bits = {.b = 0x00U, .d = 0x05U, .f = 0x00U},
+      .vdev_ops = &vmcs9900_ops,
+      .vbar_base[0] = 0x80003000,
+      .vbar_base[1] = 0x80004000,
+      .vuart_idx = 1,
+      .t_vuart.vm_id = 1U,
+      .t_vuart.vuart_id = 1U,
+    },
+
+That shows a PCI-vUART with vuart_idx=1, BDF 00:05.0, is PCI-vUART1 of this VM. We can use BDF to identify PCI-vUART in VM. Usually, legacy vuart[0] is ttyS0 in VM, and vuart[1] is ttyS1. So we hope PCI-vUART0 is ttyS0, PCI-VUART1 is ttyS1 ... PCI-vUART7 is ttyS7, but that is not true. If run ``dmesg | grep tty`` at VM shell, user may see:
+
+.. code-block:: none
+
+   [    1.276891] 0000:00:05.0: ttyS4 at MMIO 0xa1414000 (irq = 124, base_baud = 115200) is a 16550A
+
+So we know for guest OS, ttyS4 has BDF 00:05.0, is PCI-vUART1.
+
+
+if VM0 & VM1 are pre-launched VM, or SOS VM, ACRN hypervisor will create PCI-vUART virtual devices automatically. For post-launched VM, which is created by acrn-dm, additional acrn-dm option is needed to create PCI-vUART virtual device:
+
+.. code-block:: none
+
+   -s <slot>,uart,vuart_idx:<val>
+
+Kernel Config for Legacy vUART
+==============================
+
+When ACRN hypervusor passthough local APIC to a VM, there is IRQ injection issue for legacy vUART. The kernel driver should work in polling mode to avoid the problem.
```

Comment:
```suggestion
When ACRN hypervisor passthrough local APIC to a VM, there is IRQ injection issue for legacy vUART. The kernel driver should work in polling mode to avoid the problem.
```

Reply:
Will fix it